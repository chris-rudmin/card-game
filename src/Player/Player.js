import React from 'react';
import { Box, Typography, Paper, LinearProgress, Grid } from '@mui/material';
import FavoriteIcon from '@mui/icons-material/Favorite';


export default function Player({ playerNumber, playerHealth, playerDraw }) {
  return (
    <Paper elevation={3} sx={{ padding: '1em' }}>
      <Typography variant="h6">Player {playerNumber}</Typography>
      <Grid container spacing={2}>
        <Grid item xs={1}>
          <FavoriteIcon sx={{ color: 'red' }} fontSize="small"/>
        </Grid>
        <Grid item xs={11}>
          <LinearProgress variant="determinate" value={playerHealth} sx={{
            height: 10,
            borderRadius: 5,
            marginTop: '0.4em',
          }} />
        </Grid>
      </Grid>
      <Box
        direction="column"
        alignItems="center"
        justifyContent="center"
        sx={{
          width: '13em',
          height: '16em',
          border: '0.25em solid lightGrey',
          borderRadius: '0.3em',
          marginTop: '2em',
          marginBottom: '1em',
          marginLeft: 'auto',
          marginRight: 'auto',
        }}
      >
        {playerDraw}
      </Box>
    </Paper>
  );
}
