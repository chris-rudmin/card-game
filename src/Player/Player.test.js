import { render } from '@testing-library/react';
import Player from './Player';

test('should render', () => {
  const { asFragment } = render(<Player />);
  expect(asFragment()).toMatchSnapshot();
});
