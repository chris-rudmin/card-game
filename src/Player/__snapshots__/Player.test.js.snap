// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-1ce4axg-MuiPaper-root"
  >
    <h6
      class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
    >
      Player 
    </h6>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-1fdpgu7-MuiGrid-root"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-nmy1ir-MuiSvgIcon-root"
          data-testid="FavoriteIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m12 21.35-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
          />
        </svg>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-11 css-1ghyu43-MuiGrid-root"
      >
        <span
          aria-valuemax="100"
          aria-valuemin="0"
          aria-valuenow="100"
          class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate css-ro1qlq-MuiLinearProgress-root"
          role="progressbar"
        >
          <span
            class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate css-5xe99f-MuiLinearProgress-bar1"
            style="transform: translateX(0%);"
          />
        </span>
      </div>
    </div>
    <div
      class="MuiBox-root css-amhqb"
      direction="column"
    />
  </div>
</DocumentFragment>
`;
